src/components/markdown-renderer/basic-markdown-renderer.tsx

```diff
@@ -1,71 +1,45 @@
 /*
-SPDX-FileCopyrightText: 2021 The HedgeDoc developers (see AUTHORS file)
-
-SPDX-License-Identifier: AGPL-3.0-only
-*/
+ * SPDX-FileCopyrightText: 2021 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
 
 import MarkdownIt from 'markdown-it'
-import React, { ReactElement, RefObject, useMemo, useRef } from 'react'
-import { Alert } from 'react-bootstrap'
-import ReactHtmlParser from 'react-html-parser'
-import { Trans } from 'react-i18next'
+import React, { RefObject, useMemo } from 'react'
 import { useSelector } from 'react-redux'
 import { ApplicationState } from '../../redux'
-import { ShowIf } from '../common/show-if/show-if'
+import { DocumentLengthLimitReachedAlert } from './document-length-limit-reached-alert'
+import { useConvertMarkdownToReactDom } from './hooks/use-convert-markdown-to-react-dom'
 import './markdown-renderer.scss'
 import { ComponentReplacer } from './replace-components/ComponentReplacer'
-import { AdditionalMarkdownRendererProps, LineKeys } from './types'
-import { buildTransformer } from './utils/html-react-transformer'
-import { calculateNewLineNumberMapping } from './utils/line-number-mapping'
+import { AdditionalMarkdownRendererProps } from './types'
 
 export interface BasicMarkdownRendererProps {
   componentReplacers?: () => ComponentReplacer[],
   markdownIt: MarkdownIt,
   documentReference?: RefObject<HTMLDivElement>
-  onBeforeRendering?: () => void
+  onPreRendering?: () => void
```

Comment:
I actually thought about this naming while implementing that interface. I decided for "before" because the method name `onPreRendering` could be mismatched with the verb (to) prerender, but we don't do any prerenderings here.